src/table/src/metadata.rs

```diff
@@ -140,6 +141,13 @@ impl TableMetaBuilder {
     }
 }
 
+struct SplitResult<'a> {
+    columns_at_first: Vec<&'a AddColumnRequest>,
+    columns_location_map: HashMap<String, Vec<&'a AddColumnRequest>>,
```

Comment:
Suggest to rename `columns_location_map` to `columns_at_after`. New name may be odd, but consistent with the other two, easier to understand on the contrary.